qiskit_experiments/library/driven_freq_tuning/coefficient.py

```diff
@@ -0,0 +1,278 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Coefficients characterizing Stark shift."""
+
+from __future__ import annotations
+from typing import Any
+
+import numpy as np
+
+from qiskit.providers.backend import Backend
+from qiskit_ibm_experiment.service import IBMExperimentService
+from qiskit_ibm_experiment.exceptions import IBMApiError
+
+from qiskit_experiments.framework.json import ExperimentDecoder
+from qiskit_experiments.framework.backend_data import BackendData
+from qiskit_experiments.framework.experiment_data import ExperimentData
+
+
+class StarkCoefficients:
```

Comment:
Just curious, why did you switch away from a dataclass?

Reply:
When `is_dataclass(my_object) == True` I think the object is a simple data container, however, with the refactoring the `StarkCoefficients` gained several methods that perform more than data formatting. To avoid implementing something that doesn't compatible with dataclass, I decided to convert this into a subclass of bare python object.

Reply:
I see. That is reasonable. I think there are different perceptions of `dataclass` in the Python community. Some see it as a simple data container like you mention. Others see it as a helper that removes the boilerplate of writing a `self.x = x` init method and writing repr and equality methods from otherwise normal classes.